nav2_loopback_sim/nav2_loopback_sim/utils.py

```diff
@@ -63,3 +65,21 @@ def matrixToTransform(matrix):
     transform.rotation.z = quaternion[2]
     transform.rotation.w = quaternion[3]
     return transform
+
+
+def get_world2map_coordinates(world_x, world_y, map_msg):
+    map_x = int(math.floor((world_x - map_msg.info.origin.position.x) / map_msg.info.resolution))
+    map_y = int(math.floor((world_y - map_msg.info.origin.position.y) / map_msg.info.resolution))
+    return map_x, map_y
+
+
+def get_map2world_coordinates(map_x, map_y, map_msg):
```

Comment:
This appears to me to incorrect? https://github.com/ros-navigation/navigation2/blob/3f311124e356dbfa33f35e23caa5474a57f99ab7/nav2_costmap_2d/src/costmap_2d.cpp#L278-L282

Reply:
We have reference implementations: https://github.com/ros-navigation/navigation2/blob/main/nav2_simple_commander/nav2_simple_commander/costmap_2d.py#L143-L186

Reply:
Both were same but I have changed it to be same as reference